diff --git a/psplash-colors.h b/psplash-colors.h
index d701089..26df42f 100644
--- a/psplash-colors.h
+++ b/psplash-colors.h
@@ -25,10 +25,4 @@
 /* This is the color of any text output */
 #define PSPLASH_TEXT_COLOR 0x6d,0x6d,0x70
 
-/* This is the color of the progress bar indicator */
-#define PSPLASH_BAR_COLOR 0x6d,0x6d,0x70
-
-/* This is the color of the progress bar background */
-#define PSPLASH_BAR_BACKGROUND_COLOR 0xec,0xec,0xe1
-
 #endif
diff --git a/psplash-config.h b/psplash-config.h
index 82bb76d..e30099e 100644
--- a/psplash-config.h
+++ b/psplash-config.h
@@ -19,16 +19,4 @@
 #ifndef _HAVE_PSPLASH_CONFIG_H
 #define _HAVE_PSPLASH_CONFIG_H
 
-/* Text to output on program start; if undefined, output nothing */
-#define PSPLASH_STARTUP_MSG ""
-
-/* Bool indicating if the image is fullscreen, as opposed to split screen */
-#define PSPLASH_IMG_FULLSCREEN 0
-
-/* Position of the image split from top edge, numerator of fraction */
-#define PSPLASH_IMG_SPLIT_NUMERATOR 5
-
-/* Position of the image split from top edge, denominator of fraction */
-#define PSPLASH_IMG_SPLIT_DENOMINATOR 6
-
 #endif
diff --git a/psplash-fb.c b/psplash-fb.c
index d344e5a..4b3beb8 100644
--- a/psplash-fb.c
+++ b/psplash-fb.c
@@ -99,21 +99,16 @@ attempt_to_change_pixel_format (PSplashFB *fb,
 }
 
 PSplashFB*
-psplash_fb_new (int angle, int fbdev_id)
+psplash_fb_new (int fbdev_id)
 {
   struct fb_var_screeninfo fb_var;
   struct fb_fix_screeninfo fb_fix;
+  struct mxcfb_gbl_alpha gbl_alpha;
   int                      off;
   char                     fbdev[9] = "/dev/fb0";
 
   PSplashFB *fb = NULL;
 
-  if (fbdev_id > 0 && fbdev_id < 10)
-    {
-        // Conversion from integer to ascii.
-        fbdev[7] = fbdev_id + 48;
-    }
-
   if ((fb = malloc (sizeof(PSplashFB))) == NULL)
     {
       perror ("Error no memory");
@@ -124,6 +119,7 @@ psplash_fb_new (int angle, int fbdev_id)
 
   fb->fd = -1;
 
+  /* Open FBO to get info */
   if ((fb->fd = open (fbdev, O_RDWR)) < 0)
     {
       fprintf(stderr,
@@ -136,6 +132,36 @@ psplash_fb_new (int angle, int fbdev_id)
     {
       perror ("Error getting variable framebuffer info");
       goto fail;
+    }    
+
+  /* NI: Switch to selected framebuffer */
+  if (fbdev_id > 0 && fbdev_id < 10)
+    {
+        // Conversion from integer to ascii.
+        fbdev[7] = fbdev_id + 48;
+    }
+
+  fb->fd = -1;
+
+  if ((fb->fd = open (fbdev, O_RDWR)) < 0)
+    {
+      fprintf(stderr,
+              "Error opening %s\n",
+              fbdev);
+      goto fail;
+    }
+  
+  /* NI: Copy FBO variable info into selected frambuffer */
+  if (ioctl (fb->fd, FBIOPUT_VSCREENINFO, &fb_var) == -1)
+    {
+      perror ("Error getting variable framebuffer info");
+      goto fail;
+    }
+
+  if (ioctl (fb->fd, FBIOGET_VSCREENINFO, &fb_var) == -1)
+    {
+      perror ("Error getting variable framebuffer info");
+      goto fail;
     }
 
   if (fb_var.bits_per_pixel < 16)
@@ -154,6 +180,23 @@ psplash_fb_new (int angle, int fbdev_id)
       goto fail;
     }
 
+  /* NI: set alpha */
+   gbl_alpha.enable = 1;
+   gbl_alpha.alpha = 255;
+
+   if (ioctl(fb->fd, MXCFB_SET_GBL_ALPHA, &gbl_alpha) == -1)
+    {
+      perror ("Error setting alpha level");
+      goto fail;
+    } 
+
+  /* NI: unblank framebuffer */
+  if (ioctl(fb->fd, FBIOBLANK, FB_BLANK_UNBLANK) == -1)
+    {
+      perror ("Error unblanking framebuffer");
+      goto fail;
+    }  
+  
   /* NB: It looks like the fbdev concept of fixed vs variable screen info is
    * broken. The line_length is part of the fixed info but it can be changed
    * if you set a new pixel format. */
@@ -239,21 +282,6 @@ psplash_fb_new (int angle, int fbdev_id)
   status = 2;
 #endif
 
-  fb->angle = angle;
-
-  switch (fb->angle)
-    {
-    case 270:
-    case 90:
-      fb->width  = fb->real_height;
-      fb->height = fb->real_width;
-      break;
-    case 180:
-    case 0:
-    default:
-      break;
-    }
-
   return fb;
 
  fail:
@@ -279,22 +307,7 @@ psplash_fb_plot_pixel (PSplashFB    *fb,
   if (x < 0 || x > fb->width-1 || y < 0 || y > fb->height-1)
     return;
 
-  switch (fb->angle)
-    {
-    case 270:
-      off = OFFSET (fb, fb->height - y - 1, x);
-      break;
-    case 180:
-      off = OFFSET (fb, fb->width - x - 1, fb->height - y - 1);
-      break;
-    case 90:
-      off = OFFSET (fb, y, fb->width - x - 1);
-      break;
-    case 0:
-    default:
-      off = OFFSET (fb, x, y);
-      break;
-    }
+  off = OFFSET (fb, x, y);
 
   if (fb->rgbmode == RGB565 || fb->rgbmode == RGB888) {
     switch (fb->bpp)
diff --git a/psplash-fb.h b/psplash-fb.h
index d0dce10..315cec8 100644
--- a/psplash-fb.h
+++ b/psplash-fb.h
@@ -38,7 +38,7 @@ typedef struct PSplashFB
   char		*data;
   char		*base;
 
-  int            angle, fbdev_id;
+  int            fbdev_id;
   int            real_width, real_height;
 
   enum RGBMode   rgbmode;
@@ -55,7 +55,7 @@ void
 psplash_fb_destroy (PSplashFB *fb);
 
 PSplashFB*
-psplash_fb_new (int angle, int fbdev_id);
+psplash_fb_new (int fbdev_id);
 
 void
 psplash_fb_draw_rect (PSplashFB    *fb, 
diff --git a/psplash.c b/psplash.c
index 992e199..4c6b447 100644
--- a/psplash.c
+++ b/psplash.c
@@ -22,16 +22,8 @@
 #include "psplash-config.h"
 #include "psplash-colors.h"
 #include "psplash-poky-img.h"
-#include "psplash-bar-img.h"
 #include "radeon-font.h"
 
-#define SPLIT_LINE_POS(fb)                                  \
-	(  (fb)->height                                     \
-	 - ((  PSPLASH_IMG_SPLIT_DENOMINATOR                \
-	     - PSPLASH_IMG_SPLIT_NUMERATOR)                 \
-	    * (fb)->height / PSPLASH_IMG_SPLIT_DENOMINATOR) \
-	)
-
 void
 psplash_exit (int UNUSED(signum))
 {
@@ -40,92 +32,14 @@ psplash_exit (int UNUSED(signum))
   psplash_console_reset ();
 }
 
-void
-psplash_draw_msg (PSplashFB *fb, const char *msg)
-{
-  int w, h;
-
-  psplash_fb_text_size (&w, &h, &radeon_font, msg);
-
-  DBG("displaying '%s' %ix%i\n", msg, w, h);
-
-  /* Clear */
-
-  psplash_fb_draw_rect (fb, 
-			0, 
-			SPLIT_LINE_POS(fb) - h, 
-			fb->width,
-			h,
-			PSPLASH_BACKGROUND_COLOR);
-
-  psplash_fb_draw_text (fb,
-			(fb->width-w)/2, 
-			SPLIT_LINE_POS(fb) - h,
-			PSPLASH_TEXT_COLOR,
-			&radeon_font,
-			msg);
-}
-
-void
-psplash_draw_progress (PSplashFB *fb, int value)
-{
-  int x, y, width, height, barwidth;
-
-  /* 4 pix border */
-  x      = ((fb->width  - BAR_IMG_WIDTH)/2) + 4 ;
-  y      = SPLIT_LINE_POS(fb) + 4;
-  width  = BAR_IMG_WIDTH - 8; 
-  height = BAR_IMG_HEIGHT - 8;
-
-  if (value > 0)
-    {
-      barwidth = (CLAMP(value,0,100) * width) / 100;
-      psplash_fb_draw_rect (fb, x + barwidth, y, 
-    			width - barwidth, height,
-			PSPLASH_BAR_BACKGROUND_COLOR);
-      psplash_fb_draw_rect (fb, x, y, barwidth,
-			    height, PSPLASH_BAR_COLOR);
-    }
-  else
-    {
-      barwidth = (CLAMP(-value,0,100) * width) / 100;
-      psplash_fb_draw_rect (fb, x, y, 
-    			width - barwidth, height,
-			PSPLASH_BAR_BACKGROUND_COLOR);
-      psplash_fb_draw_rect (fb, x + width - barwidth,
-			    y, barwidth, height,
-			    PSPLASH_BAR_COLOR);
-    }
-
-  DBG("value: %i, width: %i, barwidth :%i\n", value, 
-		width, barwidth);
-}
-
 static int 
-parse_command (PSplashFB *fb, char *string)
+parse_command (char *string)
 {
-  char *command;
-
   DBG("got cmd %s", string);
 	
   if (strcmp(string,"QUIT") == 0)
     return 1;
 
-  command = strtok(string," ");
-
-  if (!strcmp(command,"PROGRESS")) 
-    {
-      psplash_draw_progress (fb, atoi(strtok(NULL,"\0")));
-    } 
-  else if (!strcmp(command,"MSG")) 
-    {
-      psplash_draw_msg (fb, strtok(NULL,"\0"));
-    } 
-  else if (!strcmp(command,"QUIT")) 
-    {
-      return 1;
-    }
-
   return 0;
 }
 
@@ -175,14 +89,14 @@ psplash_main (PSplashFB *fb, int pipe_fd, int timeout)
       
       if (command[length-1] == '\0') 
 	{
-	  if (parse_command(fb, command))
+	  if (parse_command(command))
 	    return;
 	  length = 0;
 	} 
       else if (command[length-1] == '\n') 
 	{
 	  command[length-1] = '\0';
-	  if (parse_command(fb, command))
+	  if (parse_command(command))
 	    return;
 	  length = 0;
 	} 
@@ -205,7 +119,7 @@ int
 main (int argc, char** argv) 
 {
   char      *tmpdir;
-  int        pipe_fd, i = 0, angle = 0, fbdev_id = 0, ret = 0;
+  int        pipe_fd, i = 0, fbdev_id = 0, ret = 0;
   PSplashFB *fb;
   bool       disable_console_switch = FALSE;
 
@@ -220,13 +134,6 @@ main (int argc, char** argv)
         continue;
       }
 
-    if (!strcmp(argv[i],"-a") || !strcmp(argv[i],"--angle"))
-      {
-        if (++i >= argc) goto fail;
-        angle = atoi(argv[i]);
-        continue;
-      }
-
     if (!strcmp(argv[i],"-f") || !strcmp(argv[i],"--fbdev"))
       {
         if (++i >= argc) goto fail;
@@ -236,7 +143,7 @@ main (int argc, char** argv)
 
     fail:
       fprintf(stderr, 
-              "Usage: %s [-n|--no-console-switch][-a|--angle <0|90|180|270>][-f|--fbdev <0..9>]\n", 
+              "Usage: %s [-n|--no-console-switch][-f|--fbdev <0..9>]\n", 
               argv[0]);
       exit(-1);
   }
@@ -268,47 +175,22 @@ main (int argc, char** argv)
   if (!disable_console_switch)
     psplash_console_switch ();
 
-  if ((fb = psplash_fb_new(angle,fbdev_id)) == NULL)
+  if ((fb = psplash_fb_new(fbdev_id)) == NULL)
     {
 	  ret = -1;
 	  goto fb_fail;
     }
 
-  /* Clear the background with #ecece1 */
-  psplash_fb_draw_rect (fb, 0, 0, fb->width, fb->height,
-                        PSPLASH_BACKGROUND_COLOR);
-
   /* Draw the Poky logo  */
   psplash_fb_draw_image (fb, 
 			 (fb->width  - POKY_IMG_WIDTH)/2, 
-#if PSPLASH_IMG_FULLSCREEN
 			 (fb->height - POKY_IMG_HEIGHT)/2,
-#else
-			 (fb->height * PSPLASH_IMG_SPLIT_NUMERATOR
-			  / PSPLASH_IMG_SPLIT_DENOMINATOR - POKY_IMG_HEIGHT)/2,
-#endif
 			 POKY_IMG_WIDTH,
 			 POKY_IMG_HEIGHT,
 			 POKY_IMG_BYTES_PER_PIXEL,
 			 POKY_IMG_ROWSTRIDE,
 			 POKY_IMG_RLE_PIXEL_DATA);
 
-  /* Draw progress bar border */
-  psplash_fb_draw_image (fb, 
-			 (fb->width  - BAR_IMG_WIDTH)/2, 
-			 SPLIT_LINE_POS(fb),
-			 BAR_IMG_WIDTH,
-			 BAR_IMG_HEIGHT,
-			 BAR_IMG_BYTES_PER_PIXEL,
-			 BAR_IMG_ROWSTRIDE,
-			 BAR_IMG_RLE_PIXEL_DATA);
-
-  psplash_draw_progress (fb, 0);
-
-#ifdef PSPLASH_STARTUP_MSG
-  psplash_draw_msg (fb, PSPLASH_STARTUP_MSG);
-#endif
-
   psplash_main (fb, pipe_fd, 0);
 
 
diff --git a/psplash.h b/psplash.h
index 7b1298f..2cc4e7e 100644
--- a/psplash.h
+++ b/psplash.h
@@ -26,6 +26,7 @@
 #include <linux/fb.h>
 #include <linux/kd.h>
 #include <linux/vt.h>
+#include <linux/mxcfb.h>
 #include <signal.h>
 #include <stdarg.h>
 #include <stdio.h>
